### PR TITLE
Make sure you can view your moderated deleted and removed communities.

### LINKED ISSUE #4911: Deleted communities can't be restored after page reload
### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

If a user accesses a deleted community page they don't see the undelete button.
This button is only shown right after deleting a community, but a fresh load of the community page does not show it.

### Steps to Reproduce

1. Create community
2. Delete community
3. Refresh
4. No restore button

### Technical Details

image
image
image

### Lemmy Instance Version

0.19.5

### Lemmy Instance URL

https://voyager.lemmy.ml

### PR DESCRIPTION
**Pull request by dessalines**
- The front end checks to see whether you are a mod, in order to be able to restore deleted / removed communities. This removes a filter which prevents that.
- Fixes #4911

### CODE REVIEW

**Review by SleeplessOne1917 — APPROVED** [review]

**Review by dullbananas — CHANGES_REQUESTED** [review]
This change appears to make it possible to see someone else's deleted communities using the read person API, which some people won't like. It's possible to fix this if `is_authenticated: bool` is replaced with `my_local_user: Option<&LocalUser>`.

**Comment by dessalines** [comment]
K fixed. It adds some annoying complication, but it works.

File: crates/db_views_actor/src/community_moderator_view.rs [hunk 1]
```diff
@@ -57,23 +59,33 @@ impl CommunityModeratorView {
       .await
   }
 
-  pub async fn for_person(
+  pub async fn for_person<'a>(
     pool: &mut DbPool<'_>,
     person_id: PersonId,
-    is_authenticated: bool,
+    local_user: Option<&'a LocalUser>,
```
Comment (dullbananas) [inline crates/db_views_actor/src/community_moderator_view.rs:65]: Try removing the 'a lifetime. It should not be needed

File: crates/db_views_actor/src/community_moderator_view.rs [hunk 2]
```diff
@@ -57,23 +59,33 @@ impl CommunityModeratorView {
...
       .inner_join(community::table)
       .inner_join(person::table)
       .filter(community_moderator::person_id.eq(person_id))
-      .filter(community::deleted.eq(false))
-      .filter(community::removed.eq(false))
       .select((community::all_columns, person::all_columns))
       .into_boxed();
-    if !is_authenticated {
+
+    if local_user.is_none() {
```
Comment (dullbananas) [inline crates/db_views_actor/src/community_moderator_view.rs:75]: Use the LocalUserOptionHelper::visible_communities_only method

Comment (dessalines) [reply crates/db_views_actor/src/community_moderator_view.rs:75]: Good call.

**Review by Nutomic — APPROVED** [review]

**Review by dullbananas — APPROVED** [review]

**Pull request merged**